ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/submission/FooDivulgerCommandGenerator.scala

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.api.benchtool.submission
+
+import com.daml.ledger.api.v1.commands.Command
+import com.daml.ledger.api.v1.value.Value
+import com.daml.ledger.client.binding.Primitive
+import com.daml.ledger.test.model.Foo.Divulger
+
+final class FooDivulgerCommandGenerator {
+
+  /** Builds a create Divulger command for each non-empty subset of divulgees
+    * such that the created Divulger contract can be used to divulge (by immediate divulgence) Foo1, Foo2 or Foo3 contracts
+    * to the corresponding subset of divulgees.
+    *
+    * @param allDivulgees - Small number of divulgees. At most 5.
+    * @return A tuple of:
+    *          - a sequence of create Divulger commands,
+    *          - a map from sorted divulgees lists (all non-empty subsets of all divulgees) to corresponding contract keys,
+    */
+  def makeCreateDivulgerCommands(
+      divulgingParty: Primitive.Party,
+      allDivulgees: List[Primitive.Party],
+  ): (List[Command], Map[List[Primitive.Party], Value]) = {
+    require(
+      allDivulgees.size <= 5,
+      s"Number of divulgee parties must be at most 5, was: ${allDivulgees.size}.",
+    )
+    def allNonEmptySubsets(divulgees: List[Primitive.Party]): List[List[Primitive.Party]] = {
```

Comment:
I notice you need to keep the parties sorted.  Would using a Set as the key solve this problem ?

Reply:
Yes!